### DISCUSSION TITLE
Local Storage persistence

### DISCUSSION THREAD

**Discussion by AdaptGit** [ionic-team/capacitor#3321] [Q&A]
Hi 

I don't know if i am doing something wrong however i'm using the following this article 
https://capacitorjs.com/docs/apis/storage

i have an pwa/android/ios that uses single sign on, and stores the access information in local storage. the app works fine. however when they close the app on android or ios it doesn't automatically login when the relaunch the app. it works fine from the website and pwa. 

what i think is happening is when the user closes the app on android or IOS the local storage is cleared.

Has anyone else had a similar problem? or am using the wrong plugin? and help will be appreciated

**Comment by dasturchioka** [comment]
Hi

I am having the same issue, did you find the solution by chance?

**Comment by rricamar** [comment]
@AdaptGit, @dasturchioka from the Capacitor Documentation itself ([Storage doc](https://capacitorjs.com/docs/guides/storage)):

> Local Storage can be used for small amounts of temporary data, such as a user id, **but must be considered transient**, meaning your app needs to expect that the data will be lost eventually. This is because the OS will reclaim local storage from Web Views if a device is running low on space.

Maybe you can take a look to [@capacitor-community/sqlite](https://github.com/capacitor-community/sqlite) plugin or [capacitor-secure-storage](https://github.com/aparajita/capacitor-secure-storage) plugin 👍

**Reply by dasturchioka** [reply]
Thank you for the solution, but my problem was not with LocalStorage persistance. After Iʼve read my whole codesource, Iʼve implemented a function that resets the whole storage, and ran it immediately.
